core/src/main/scala/org/bitcoins/core/protocol/tlv/DLCOutcomeType.scala

```diff
@@ -0,0 +1,20 @@
+package org.bitcoins.core.protocol.tlv
+
+import org.bitcoins.crypto.CryptoUtil
+import scodec.bits.ByteVector
+
+sealed trait DLCOutcomeType {
+  def serialized: Vector[ByteVector]
+}
+
+case class EnumOutcome(outcome: String) extends DLCOutcomeType {
+
+  override def serialized: Vector[ByteVector] =
+    Vector(CryptoUtil.serializeForHash(outcome))
+}
+
+case class UnsignedNumericOutcome(digits: Vector[Int]) extends DLCOutcomeType {
+
```

Comment:
should require all are positive

Reply:
I'm worried that this would harm efficiency and doesn't really provide much benefit as everything else will break given negative digits. Too bad Java doesn't have a UInt type like rust :(